app-shells/hstr/patches/hstr-1.24.patchset

```diff
@@ -0,0 +1,86 @@
+From 419c19184f486d849f845e8485fd552d4c83b51b Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Sun, 29 Apr 2018 12:58:42 +0200
+Subject: Haiku supporting patches
+
+
+diff --git a/src/Makefile.am b/src/Makefile.am
+index c3e0823..20f8e44 100644
+--- a/src/Makefile.am
++++ b/src/Makefile.am
+@@ -34,6 +34,6 @@ hh_SOURCES = 						\
+ 
+ # create hh > hstr hard link on installation
+ install-exec-hook:
+-	ln $(DESTDIR)$(bindir)/hh$(EXEEXT) $(DESTDIR)$(bindir)/hstr$(EXEEXT)
++	ln -s $(DESTDIR)$(bindir)/hh$(EXEEXT) $(DESTDIR)$(bindir)/hstr$(EXEEXT)
+ 
+ hh_LDADD = $(NCURSES_LIBS)
+diff --git a/src/hstr.c b/src/hstr.c
+index 80b18d2..5141663 100644
+--- a/src/hstr.c
++++ b/src/hstr.c
+@@ -22,6 +22,8 @@
+ #include <locale.h>
+ #ifdef __APPLE__
+ #include <curses.h>
++#elif defined(__HAIKU__)
++#include <ncurses.h>
+ #elif defined(__FreeBSD__)
```

Comment:
You could do an `||` with the `defined(__FreeBSD__)` below as well, which IMO, would be cleaner

Reply:
Hmm, the diff would be _smaller_, but I wouldn't say that it would be _cleaner_.
Keeping a separate `#elif defined(__HAIKU__)` has one tiny advantage, of course: if the set of headers needed for this purpose ever changes between Haiku and FreeBSD, the "impact" would be isolated.

(If the project already had a few OSes grouped by categories (e.g. BSD*, Linux, etc...) then things would be different.)